workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -120,7 +121,7 @@ internal class WorkspaceServiceImpl(
             ownerId = getCurrentAuthenticatedUserName(csmPlatformProperties))
     createdWorkspace.setRbac(csmRbac.initSecurity(workspace.getRbac()))
 
-    return workspaceRepository.save(createdWorkspace)
+    return checkReadSecurity(workspaceRepository.save(createdWorkspace))
```

Comment:
Same remark here, if you're able to create something, you have all permission on it